massa-bootstrap/src/server.rs

```diff
@@ -603,16 +608,7 @@ pub async fn stream_bootstrap_information<D: Duplex>(
         }
 
         if slot_too_old {
-            match server.send_msg(write_timeout, BootstrapServerMessage::SlotTooOld) {
-                Err(BootstrapError::TimedOut(_)) => Err(std::io::Error::new(
-                    std::io::ErrorKind::TimedOut,
-                    "SlotTooOld message send timed out",
-                )
-                .into()),
-                Err(e) => Err(e),
-                Ok(_) => Ok(()),
-            }?;
-            return Ok(());
+            return server.send_msg(write_timeout, BootstrapServerMessage::SlotTooOld);
```

Comment:
This would be perfect if we could have something like 

`return server.send_msg(...).update_timeout_err("SlotTooOld message send timed out")`;

(same for other parts where the similar message setting is happening)

Reply:
close with https://github.com/massalabs/massa/pull/3763